plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4862,10 +4862,15 @@ public synchronized Network setupvSwitchNetwork(final Connection conn) {
         return null;
     }
 
-    public void shutdownVM(final Connection conn, final VM vm, final String vmName) throws XmlRpcException {
+    public void shutdownVM(final Connection conn, final VM vm, final String vmName, final boolean forcedStop) throws XmlRpcException {
         Task task = null;
         try {
-            task = vm.cleanShutdownAsync(conn);
+            if (forcedStop) {
+                task = vm.hardShutdownAsync(conn);
+            } else {
+                task = vm.cleanShutdownAsync(conn);
+            }
+
```

Comment:
The method as it was tried to do a graceful shutdown and upon failure will be followed by a hard shutdown, so in a way, the shutdownVM was taking care of the scenario in which the VM may not stop gracefully. This method does not really need a forcedStop param.